pkg/gpu/nvidia/manager.go

```diff
@@ -83,7 +81,7 @@ type MountPath struct {
 	ContainerPath string
 }
 
-func NewNvidiaGPUManager(devDirectory string, mountPaths []MountPath, gpuConfig GPUConfig) *nvidiaGPUManager {
+func NewNvidiaGPUManager(devDirectory, procDirectory string, mountPaths []MountPath, gpuConfig GPUConfig) *nvidiaGPUManager {
```

Comment:
I changed procDirectory to be an input when initializing NvidiaGPUManager, so that the unit test can directly replace this procDirectory with a test path. Changing the procDirectory to be an input doesn't change the existing logic, and it help to save effort to write a set of mock functions for overriding (otherwise, the procDirectory is unchangeable).